openlibrary/plugins/openlibrary/js/page-barcodescanner/index.js

```diff
@@ -69,3 +77,15 @@ export function init() {
 function isBarcodeISBN(code) {
     return code.startsWith('97');
 }
+
+function CheckISBN(code, confirmation_list) {
+    if (confirmation_list.length == 0) {
+        confirmation_list.push(code);
+    }
+    if (confirmation_list.includes(code)) {
+        confirmation_list.splice(0, confirmation_list.length);
+        return true;
+    }
+    else
+        return false;
```

Comment:
```suggestion
    if (confirmation_list.includes(code)) {
        return true;
    } else {
        confirmation_list.push(code);
        return false;
    }
```
It looks like this function can never return false in its current state.  Items are only being added to the array in this function, and are being immediately removed after the second `if` statement.

Removing the `splice` call will allow us to use this function to check for redundant codes when multiple scans are enabled (which I think we want to do).  In this case, we'll probably want to maintain a second array containing codes that were used to create `LazyBookCard`s.  Before creating a new card, we would ensure that `checkISBN` returns true and that the code is not present in the new array.